### ISSUE TITLE
Future schema upgrades

### ISSUE THREAD

**Issue by this-username-is-taken** [messari/subgraphs#389]
- Add strategy entity and harvest events to yield aggregators.
- Add underlying tokens to input tokens in yield aggregators.
- Add more reward token types (in Aave-v3, reward emissions are per token: aToken/stableDebtToken/variableDebtToken).

**Comment by steegecs** [comment]
- Add number of days or hours since unix timestamp as a field for entities that concatenate them to some prefix in the id (i.e. LiquidityPoolDailySnapshots).

**Comment by dmelotik** [comment]
- Lending protocol: add `positionCounts` to snapshots (market/usage)

**Comment by tnkrxyz** [comment]
Make these two fields of the Market entity in the lending protocol optional, outputToken is already optional.
- outputTokenSupply
- outputTokenPriceUSD

**Comment by steegecs** [comment]
Yield Aggregatort
- outputTokenSupply is not required in the Vault entity but required in the Vault snapshots
- UsageMetricsDaily snapshot has totalPoolCount but the hourly snapshot does not.

**Comment by dmelotik** [comment]
- transaction counts on a market-level 
- user/unique user counts on market-level

**Comment by dmelotik** [comment]
- add a field at the protocol level called "protocol" to store the name without any versions. ie, "aave" instead of "aave-v2"

**Comment by dmelotik** [comment]
- specify if `withdraw.amount / amountUSD` should include or exclude withdrawal fee if applicable.

Based on convo with @bye43 this should be excluded

**Comment by this-username-is-taken** [comment]
add oracle to lending protocols

**Comment by this-username-is-taken** [comment]
add aggregated token data info in dex subgraphs (similar to uniswap)

**Comment by dmelotik** [comment]
Consider allowing multiple input tokens in lending protocols. For the case of Compound V3 markets

**Comment by dmelotik** [comment]
Consider having rewards on the protocol level. In the case of TrueFi that is what is happening. @corerouter feel free to add any other comments or notes on this front! Thanks for bringing it up

**Comment by dmelotik** [comment]
How can we show variable vs stable debt in lending positions? prolly by separating the two into different positions, but we need schema changes to allow for this i think

**Comment by steegecs** [comment]
- For DEXs, does it make sense to have `from` and `to` fields in the swaps? A user should always be swapping with a pool, though, they may interact with a Router as a proxy. So I am thinking about removing `from` and `to` and just having `user`, `pool`, and maybe `router` in the case of swaps. I think this is more clear.

**Comment by dmelotik** [comment]
Should `positiion.liquidationCount` increment when collateral is transferred to the liquidator, and specify this in schema.

**Comment by steegecs** [comment]
- Are `schema` and `protocol type` one to one? If not, should we consider adding schema into the protocol entity?

**Comment by tnkrxyz** [comment]
Rename PositionSide.LENDER to PositionSide.Collateral in extended lending schema? Protocols like makerdao do not lend collaterals (only DAI); it is a bit odd to call them the lender side.

**Comment by tnkrxyz** [comment]
Make the liquidate.position field a list in the extended lending schema, because multiple positions will be changed by a liquidation (lender and borrower side).

**Comment by dmelotik** [comment]
Integrate flash loans into lenfing schema. Otherwise a new schema for it. Could make for some cool data apps 😉

**Comment by dmelotik** [comment]
How do we classify position token (ie, aToken / cToken) transfer? Should this be a new position, should it count as a withdraw / deposit?

**Comment by this-username-is-taken** [comment]
> How do we classify position token (ie, aToken / cToken) transfer? Should this be a new position, should it count as a withdraw / deposit?

That's correct. These is a fairly long discussion on this in the Discord if you try searching through

**Comment by dmelotik** [comment]
use immutable where possible per #964

**Comment by dmelotik** [comment]
Add native token balances where possible, this would include borrow balances in native unit. Per discussion wigth @yulesa (feel free to add more detail if necessary here

**Comment by yulesa** [comment]
Let me give examples of where this is useful.
Imagine you're analyzing a pool and it's TVL went down, but that's only beacuse the price tanked, in native terms the pool is only went up. Native terms remove price fluctuations but you can't combine different pools.

**Comment by dmelotik** [comment]
yeild ags should have `isActive` on the vaults, beefy finance has deprecated vaults for example

**Comment by this-username-is-taken** [comment]
Add entity count to make fetching large number of paginated entities easier (instead of blindly fetching until the end of pagination)
